src/main/webapp/scripts/matrixController.js

```diff
@@ -1,32 +1,179 @@
         }
-        matrix += rowClosure;
+        matrix.push(tempRow);
+        matrixToTable += rowClosure;
     }
-    document.getElementById("matrixTable").innerHTML = matrix;
+    document.getElementById("matrixTable").innerHTML = matrixToTable;
+}
+
+
+function updateMatrixCell(object){
+    let row = parseInt( object.id[0] );
+    let column = parseInt( object.id[1] );
+    switch( selectedCellType ){
+        case "deadCell":
+            object.src = "../images/dead cell.png";
+            matrix[row][column] = 0;
+            break;
+        
+        case "aliveCell":
+            if (!cellWithMask && !cellWithVaccine){
+                object.src = "../images/alive normal cell.png";
+                matrix[row][column] = 1;
+            }else if (cellWithMask && cellWithVaccine){
+                object.src = "../images/alive mask vaccine.png";
+                matrix[row][column] = 4;
+            }else if (cellWithMask){
+                object.src = "../images/alive mask.png";
+                matrix[row][column] = 2;
+            }else{
+                object.src = "../images/alive vaccine.png";
+                matrix[row][column] = 3;
+            }
+            
+            break;
+        
+        case "infectedCell":
+            if (!cellWithMask && !cellWithVaccine){
+                object.src = "../images/Infected Cell.png";
+                matrix[row][column] = 5;
+            }else if (cellWithMask && cellWithVaccine){
+                object.src = "../images/infected mask vaccine.png";
+                matrix[row][column] = 8;
+            }else if (cellWithMask){
+                object.src = "../images/infected mask.png";
+                matrix[row][column] = 6;
+            }else{
+                object.src = "../images/infected vaccine.png";
+                matrix[row][column] = 7;
+            }
+            break;
+    }
+}
+
+function updateSelectedCellType(object){
+    selectedCellType = object.id;
+    switch(selectedCellType)
+    {
+        case "deadCell":
+            object.src = "../images/dead selected.png";
+            document.getElementById("aliveCell").src = "../images/alive normal cell.png";
+            document.getElementById("infectedCell").src = "../images/Infected Cell.png";
+            break;
+        
+        case "aliveCell":
+            object.src = "../images/live selected.png";
+            document.getElementById("deadCell").src = "../images/dead cell.png";
+            document.getElementById("infectedCell").src = "../images/Infected Cell.png";
+            break;
+        
+        case "infectedCell":
+            object.src = "../images/infected selected.png";
+            document.getElementById("deadCell").src = "../images/dead cell.png";
+            document.getElementById("aliveCell").src = "../images/alive normal cell.png";
+            break;
+    }
+}
+
+function updateSpeed(object){
+    playSpeed = 1000 - ( (parseInt(object.value) - 1) * 100 );
```

Comment:
nit: Rather than representing the playSpeed as an arbitrary number, it might be more intuitive to give them direct control over the # of milliseconds between each update. https://www.w3schools.com/jsref/prop_range_step.asp seems potentially relevant if you wanted to restrict people to multiples of 100ms (like above).